starfish/segmentation_mask.py

```diff
@@ -0,0 +1,177 @@
+
+        self._masks = masks
+
+    def __getitem__(self, index):
+        return self._masks[index]
+
+    def __iter__(self):
+        return iter(self._masks)
+
+    def __len__(self):
+        return len(self._masks)
+
+    def add_mask(self, mask: xr.DataArray):
+        """Add an existing segmentation mask.
+
+        Parameters
+        ----------
+        arr : xr.DataArray
+            Segmentation mask.
+        """
+        validate_segmentation_mask(mask)
+        self._masks.append(mask)
+
+    @classmethod
+    def from_label_image(
+            cls,
+            label_image: np.ndarray,
+            physical_ticks: Dict[Coordinates, List[float]]
+    ) -> "SegmentationMaskCollection":
+        """Creates segmentation masks from a label image.
+
+        Parameters
+        ----------
+        label_image : int array
+            Integer array where each integer corresponds to a cell.
+        physical_ticks : Dict[Coordinates, List[float]]
+            Physical coordinates for each axis.
+
+        Returns
+        -------
+        masks : SegmentationMaskCollection
+            Masks generated from the label image.
+        """
+        props = regionprops(label_image)
+
+        if label_image.ndim == 2:
+            dims = AXES[1:]
+        elif label_image.ndim == 3:
+            dims = AXES
+        else:
+            raise TypeError('expected 2- or 3-D image')
+
+        masks: List[xr.DataArray] = []
+
+        coords: Dict[str, Union[list, Tuple[str, list]]]
+
+        for label, prop in enumerate(props):
+            coords = {d: list(range(prop.bbox[i], prop.bbox[i + len(dims)]))
+                      for i, d in enumerate(dims)}
+
+            for d, c in physical_ticks.items():
+                axis = d.value[0]
+                i = dims.index(axis)
+                coords[d.value] = (axis, c[prop.bbox[i]:prop.bbox[i + len(dims)]])
+
+            mask = xr.DataArray(prop.image,
+                                dims=dims,
+                                coords=coords,
+                                name=str(label + 1))
+            masks.append(mask)
+
+        return cls(masks)
+
+    @classmethod
+    def from_disk(cls, path: str) -> "SegmentationMaskCollection":
+        """Load the collection from disk.
+
+        Parameters
+        ----------
+        path : str
+            Path of the directory to instantiate from.
+
+        Returns
+        -------
+        masks : SegmentationMaskCollection
+            Collection of segmentation masks.
+        """
+        masks = []
+        for p in os.listdir(path):
+            mask = xr.open_dataarray(osp.join(path, p))
+            masks.append(mask)
+
+        return cls(masks)
+
+    def save(self, path: str, overwrite: bool = False):
+        """Save the segmentation masks to disk.
+
+        Parameters
+        ----------
+        path : str
+            Path of the directory to write to.
+        overwrite : bool, optional
```

Comment:
I think the pattern used thus far in this repo is to do:

```
overwrite : bool
    Whether to overwrite the directory if it exists. (default: False)
```

If you don't feel strongly about it, let's stick with this convention.  If you do feel strongly about it, let's discuss. :)